tool/templates/kotlin/Opaque.kt.jinja

```diff
@@ -40,11 +40,20 @@ class {{type_name}} internal constructor (
 {%- endfor %}
 {%- endif %} {
 
+    {%- if !use_finalizers_instead_of_cleaners %}
+
     internal class {{type_name}}Cleaner(val handle: Pointer, val lib: {{type_name}}Lib) : Runnable {
         override fun run() {
             lib.{{dtor_abi_name}}(handle)
         }
     }
+    {%- else %}
+    @Override
+    @SuppressWarnings("Finalize")
+    fun finalize() {
```

Comment:
wait kotlin used to let you just tack on a finalize method? i did _not_ expect it to be that old school.

no wonder it's deprecated.

Reply:
yeah, i think the warning is because it's deprecated now -- you can do the same thing in Java

Reply:
Yeah, I'm familiar with the Java thing I had just almost completely memoryholed it because it's like, something I remember from when I first encountered Java as a kid.

So it was a bit of a throwback seeing this :)